netebpfext/net_ebpf_ext_hook_provider.c

```diff
@@ -194,33 +142,26 @@ _net_ebpf_extension_detach_client_completion(_In_ PDEVICE_OBJECT device_object,
     work_item = hook_client->detach_work_item;
 
     // Wait for any in progress callbacks to complete.
-    _ebpf_ext_attach_wait_for_rundown(&hook_client->rundown, hook_client->execution_type);
+    _ebpf_ext_attach_wait_for_rundown(&hook_client->rundown);
 
     // Note: This frees the provider binding context (hook_client).
     NmrProviderDetachClientComplete(hook_client->nmr_binding_handle);
 
     IoFreeWorkItem(work_item);
 }
 
-_Acquires_lock_(hook_client) bool net_ebpf_extension_hook_client_enter_rundown(
-    _Inout_ net_ebpf_extension_hook_client_t* hook_client, net_ebpf_extension_hook_execution_t execution_type)
+bool
+net_ebpf_extension_hook_client_enter_rundown(_Inout_ net_ebpf_extension_hook_client_t* hook_client)
 {
     net_ebpf_ext_hook_client_rundown_t* rundown = &hook_client->rundown;
-    if (execution_type == EXECUTION_PASSIVE) {
-        ACQUIRE_PUSH_LOCK_SHARED(&rundown->passive.lock);
-    }
-
-    return (rundown->rundown_occurred == FALSE);
+    return (ExAcquireRundownProtection(&rundown->protection));
```

Comment:
```suggestion
    return ExAcquireRundownProtection(&rundown->protection);
```
nit for better readability and consistency with line 164